python/ray/serve/api.py

```diff
@@ -571,3 +577,153 @@ def delete(name: str, _blocking: bool = True):
     """
     client = get_global_client()
     client.delete_apps([name], blocking=_blocking)
+
+
+@PublicAPI(stability="alpha")
+def multiplexed(
+    func: Optional[Callable[..., Any]] = None, max_num_models_per_replica: int = -1
+):
+    """[EXPERIMENTAL] Coverts a function or method to a multiplexed function.
+
+    The function can be standalone function or a method of a class. The
+    function must have exact one argument with type `str` and the argument
+    must be the first argument. The `str` argument is the model id of the model to
+    be loaded.
+
+    User have to define the function with `async def` and the function must be
+    an async function. It is recommended to define courotine for long running
+    IO tasks in the function, it can prevent blocking the event loop.
+
+    The multiplexed function is called to load a model with a given model ID when
+    necessary.
+
+    When the number of models in one replica is larger than max_num_models_per_replica,
+    the multiplexed function will unload the model with LRU policy.
+
+    If you want to release resource after the model is loaded, you can define
+    a `__del__` method in your model class. The `__del__` method will be called when
+    the model is unloaded.
+
+    Example:
+
+    .. code-block:: python
+            from ray import serve
+
+            @serve.deployment
+            class MultiplexedDeployment:
```

Comment:
suggest to add constructor that sets some "s3 path" or something (to show that you can still do that)

Reply:
Hi @edoakes added a constructor for it in the constructor.